### PR TITLE
this is the block of code for the printWordList

### PR DESCRIPTION
**Pull request by vamcabalo**
This reveals all the found and possible answers of the user. The minimum number of letters per word is set to 3.

### CODE REVIEW

File: word-search.py [hunk 1]
```diff
@@ -229,8 +229,39 @@ def checkWord():
     # checks if user found a correct word
     return
 
-def printWordList():
-    # prints all possible words to be found at game end
+def printWordList(gridWordList, foundWords):
+    # Prints all possible words to be found at game end
+    # Reveals player's answers and possible answers
+    
+    from collections import defaultdict
```
Comment (vdepasupillagazo) [inline word-search.py]: Import statements should always be declared at the top of the file. Please move to line 8

File: word-search.py [hunk 2]
```diff
@@ -229,8 +229,39 @@ def checkWord():
     # checks if user found a correct word
     return
 
-def printWordList():
-    # prints all possible words to be found at game end
+def printWordList(gridWordList, foundWords):
```
Comment (vdepasupillagazo) [inline word-search.py]: Please update name to print_word_list to follow our conventions

File: word-search.py [hunk 3]
```diff
@@ -229,8 +229,39 @@ def checkWord():
...
+    # Prints all possible words to be found at game end
+    # Reveals player's answers and possible answers
+    
+    from collections import defaultdict
+
+    # Group words by their lengths from gridWordList
+    length_groups = defaultdict(list)
+    for word in gridWordList:
+        if len(word) >= 3:  # Only consider words with length 3 or more
+            length_groups[len(word)].append(word)
```
Comment (vdepasupillagazo) [inline word-search.py]: We already filter for word length in the load_word_library function. Please remove 😄

File: word-search.py [hunk 4]
```diff
@@ -229,8 +229,39 @@ def checkWord():
...
+    possible_words = defaultdict(list)
+    for length in length_groups:
+        for word in length_groups[length]:
+            if word not in foundWords:
+                possible_words[length].append(word)
+
+    # Print possible words
+    print("\nHere are the possible words:")
+    for length in possible_words:
+        print(" ".join(possible_words[length]))
```
Comment (vdepasupillagazo) [inline word-search.py]: This is good, but can we also include a header/label to indicate which list we are printing? For example:
Screenshot 2024-12-10 at 3 34 23 PM

Just to make the distinction clearer to the user 😄

File: word-search.py [hunk 5]
```diff
@@ -229,8 +229,39 @@ def checkWord():
...
+
+    # Print answered (found) words
+    print("\nHere are the words you have found:")
+    found_words_str = " ".join(foundWords)
+    print(found_words_str)
```
Comment (vdepasupillagazo) [inline word-search.py]: Can we also apply the separation per word length to the found words list?

**Review by vdepasupillagazo — CHANGES_REQUESTED** [review]

File: word-search.py [hunk 6]
```diff
@@ -5,6 +5,7 @@
 import shutil  #Module for getting terminal dimensions
 import random
 import re      
+from collections import defaultdict
```
Comment (vamcabalo) [inline word-search.py:8]: @vickydepasupil already moved the import library at line 8 :))

Comment (vdepasupillagazo) [reply word-search.py:8]: Thank you!

**Comment by vdepasupillagazo** [comment]
Closing this PR. A new branch has been created for the print word list function.

**Pull request closed without merging**